pkg/controller/ipam/antrea_ipam_controller.go

```diff
@@ -256,11 +261,12 @@ func (c *AntreaIPAMController) cleanIPPoolForStatefulSet(namespacedName string)
 		if err != nil {
 			// This can be a transient error - worker will retry
 			klog.ErrorS(err, "Failed to clean IP allocations", "StatefulSet", namespacedName, "IPPool", ipPool.Name)
+			retriableErr = err
```

Comment:
should we use an error list and https://pkg.go.dev/k8s.io/apimachinery/pkg/util/errors#NewAggregate instead? This way we can handle the multiple-errors case

Reply:
why not return a generic error then, e.g., "clean up failed for at least one IPPool"?
otherwise we will end up logging an error twice anyway

Reply:
nit: not your code but the `namespace, name := k8s.SplitNamespacedName(namespacedName)` line above can be moved outside of the loop

Reply:
Updated.